### PR TITLE
[hotfix] flutter create -t plugin remove no-op iOS folder and set sdk min to 1.20.0

### LINKED ISSUE #60215: `flutter create -t plugin` should not create no-op iOS folders in 1.20.0 stable release
When 1.20 releases, we should make `flutter create -t plugin` command not creating no-op iOS folders anymore.
Also, we will add a lower flutter SDK bound setting to 1.20.0 for all the newly created plugins. 

Ideally, we'd like to land this change to the 1.20.0 release branch as it goes out.

### PR DESCRIPTION
**Pull request by cyanglaz**
This is a cherry-pick of https://github.com/flutter/flutter/commit/004f90f8fd8a8fb93d386d55364707ef1e517948 and https://github.com/flutter/flutter/commit/2459f5c513e26f7bc60d68d8e58972d37bbd6e58

Fixes https://github.com/flutter/flutter/issues/60215

### CODE REVIEW

**Comment by cyanglaz** [comment]
@pcsosinski @csells This is the hotfix PR.
When the new hotfix branch is created, I'll update the target branch.

**Comment by cyanglaz** [comment]
I think we should cherry-pick https://github.com/flutter/flutter/pull/62630 in as well, which adds more tests. So we can be more confident. @jmagman @pcsosinski WDYT?

File: packages/flutter_tools/test/commands.shard/permeable/create_test.dart [hunk 1]
```diff
@@ -1542,7 +1543,7 @@ void main() {
 
     // TODO(cyanglaz): no-op iOS folder should be removed after 1.20.0 release
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/permeable/create_test.dart:1544]: remove TODO?

**Review by christopherfujino — APPROVED** [review]
LGTM with a nit about removing a TODO comment.

**Comment by cyanglaz** [comment]
Experienced some issue with tags. Closing this in favor of https://github.com/flutter/flutter/pull/62990

**Pull request closed without merging**